### ISSUE TITLE
No OpKernel was registered to support Op 'FIFOQueueV2' with these attrs.

### ISSUE THREAD

**Issue by derekhh** [tensorflow/tensorflow#8454]
### Description

I'm trying to use the [Inception-ResNet-V2](https://github.com/tensorflow/models/tree/master/slim) model from TF-slim in the Android demo app. I'm now seeing the following inference exception:

    E/TensorFlowInferenceInterface: Inference exception: java.lang.IllegalArgumentException: No OpKernel was registered to support Op 'FIFOQueueV2' with these attrs.  Registered devices: [CPU], Registered kernels:
                                                                                     <no registered kernels>
                                                                                   
                                                                                   	 [[Node: batch/fifo_queue = FIFOQueueV2[capacity=128, component_types=[DT_FLOAT, DT_UINT8, DT_INT64], container="", shapes=[[299,299,3], [299,299,3], []], shared_name=""]()]]

To ensure this issue does not come from my build, I've download the latest [libtensorflow_inference.so](http://ci.tensorflow.org/view/Nightly/job/nightly-android/70/artifact/out/native/) from Jenkins. I've verified that this library still provides the same exception.

I'm not familiar with the error message but it seems the Op `FIFOQueueV2` is defined in [fifo_queue_op.cc](https://github.com/tensorflow/tensorflow/blob/4433079e7f317724eaa92ec120c6b1c3c0c52f2f/tensorflow/core/kernels/fifo_queue_op.cc#L61). Can someone help take a look?

**Comment by concretevitamin** [comment]
@nathansilberman - does it look like it's an issue of the BUILD file needing to be updated to depend on the FIFOQueueV2 kernel?

**Comment by derekhh** [comment]
@concretevitamin Thanks for your reply.

@nathansilberman: I've found a workaround by loading this model checkpoint, create a `tf.placeholder` and then use this as the `input_node` instead. 

Please correct me if I am wrong. I guess there is no easy way to use the current queue-based input graph on Android. If we use that then the input node needs to be a `batch/queue` or a `shuffle_batch/random_shuffle_queue`, but there is no supporting OpKernel for these Ops right now.

**Comment by aleksab** [comment]
@derekhh how did you get it working by simply creating a placeholder? The FIFOQueueV2 node is still there and shoudl cause a NoOp on android? I am struggling with the same issue

**Comment by derekhh** [comment]
@aleksab Sorry for not being clear.

Basically I've created a new Python script that looks like this. This restores the TF-Slim Inception model from your previous checkpoint and reads from a placeholder. The model `inception_v1_final.pb` created can be used for Android.

I'm quite new to TensorFlow so I'm sure this won't be the best solution - creating a new Python script for this purpose already seems stupid. If there's a better solution please let me know. Thanks!

    import os
    import tensorflow as tf
    import tensorflow.contrib.slim as slim

    from nets import inception
    from nets import inception_v1
    from nets import nets_factory

    from tensorflow.python.framework import graph_util
    from tensorflow.python.platform import gfile

    checkpoint_path = tf.train.latest_checkpoint('./log')

    with tf.Graph().as_default() as graph:
        input_tensor = tf.placeholder(tf.float32, shape=(None, 224, 224, 3), name='input_image')
        with tf.Session() as sess:
            with tf.variable_scope('model') as scope:
                with slim.arg_scope(inception.inception_v3_arg_scope()):
                    logits, end_points = inception.inception_v1(input_tensor, num_classes=45, is_training=False)

        saver = tf.train.Saver()
        saver.restore(sess, checkpoint_path)

        output_node_names = 'model/InceptionV1/Logits/Predictions/Softmax'
        input_graph_def = graph.as_graph_def()
        output_graph_def = graph_util.convert_variables_to_constants(sess, input_graph_def, output_node_names.split(","))
        with open('./output_graph_nodes.txt', 'w') as f:
            for node in output_graph_def.node:
                f.write(node.name + '\n')

        output_graph = './inception_v1_final.pb'
        with gfile.FastGFile(output_graph, 'wb') as f:
            f.write(output_graph_def.SerializeToString())

**Comment by girving** [comment]
Many ops are disabled on Android for code size reasons, so this is WAI unfortunately.

**Comment by nesadiankemo** [comment]
Hey, guys. 
I was encountered that problem recently. And I fix it just by put "fifo_queue_op.cc" in file 'tensorflow/core/kernels/BUILD'  inside the `filegroup(
    name = "android_core_ops", srcs = [ ...`, which configure the core kernel in Android.

**Comment by anandcu3** [comment]
@nesadiankemo  I have encountered the same issue. . but your fix did not help.. I added "fifo_queue_op.cc" in android_core_ops and also in tensorflow/contrib/makefile/tf_op_files.txt . But the same error still comes up..  Did you take any other steps to fix it. Could you please explain?

**Comment by nesadiankemo** [comment]
@anandcu3 In the source code, I only modify two source file. Another file is I commented at #2680 to fix that issue. Then I follow the instruction [https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/android](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/android) to build the .so file. And it fix the issue in my App.
Here is all the steps i had taken. Hope it will help you.
